ovs/dal/dataobject.py

```diff
@@ -679,12 +679,15 @@ def delete(self, abandon=None, _hook=None):
                     if info['list'] is True:
                         if len(items) > 0:
                             if abandon is not None and (key in abandon or '_all' in abandon):
-                                for item in items.itersafe():
-                                    setattr(item, info['key'], None)
-                                    try:
-                                        item.save()
-                                    except ObjectNotFoundException:
-                                        pass
+                                try:
+                                    for item in items.itersafe():
+                                        setattr(item, info['key'], None)
+                                        try:
+                                            item.save()
+                                        except ObjectNotFoundException:
+                                            pass
+                                except KeyNotFoundException:
```

Comment:
The issue is not only at/in the itersafe, but based on the reported stacktrace it's caused by the multi-get that got a non-existing key. It can basically happen at any place where a query is executed. The query itself should be smart enough to re-execute without the missing key.